orttraining/orttraining/training_ops/cuda/nn/layer_norm_impl.cu

```diff
@@ -504,7 +537,11 @@ void HostLayerNormGradient(
   // compute grad_input
   const uint64_t maxGridY = prop.maxGridSize[1];
   const dim3 blocks1(1, std::min<unsigned int>(static_cast<unsigned int>(n1), static_cast<unsigned int>(maxGridY)), 1);
-  const dim3 threads1(warp_size, 4, 1);
+  dim3 threads1(warp_size, 4, 1);
+#ifdef HIP_VERSION
+  // Optimization for ROCm MI100
+  threads1.y = 2;
```

Comment:
If you only change block size here, how much gain did you see? I am curious whether the change in function cuComputeGradInput is needed or not.

Reply:
We saw only marginal improvement with just changing block size for cuComputeGradInput.
![image](https://user-images.githubusercontent.com/81309834/118157890-b70c1280-b3e0-11eb-9f52-69f6c2401442.png)

Reply:
We saw only marginal improvement with just changing block size.

Reply:
Changing block size is marginal perf improvement ( < 10%), and its not uniform for all cases we tested.